pkg/webhook/server.go

```diff
@@ -79,6 +101,18 @@ type Server struct {
 
 	// WebhookMux is the multiplexer that handles different webhooks.
 	WebhookMux *http.ServeMux
+}
+
+// New constructs a new Server from the provided options.
+func New(o Options) Server {
+	return &DefaultServer{
+		Options: o,
+	}
+}
+
+// DefaultServer is the default implementation used for Server.
+type DefaultServer struct {
+	Options Options
```

Comment:
Wondering if Options should be private given that we also have a constructor (but no strong opinion)

Reply:
The reason I left it public is that there are a bunch of tests in the `Manager` that check some values in there